src/Umbraco.Web.BackOffice/Controllers/ContentController.cs

```diff
@@ -1960,6 +1960,58 @@ public IActionResult PostPublishById(int id)
         return Ok();
     }
 
+    /// <summary>
+    ///     Publishes a document with a given ID and cultures.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    /// <remarks>
+    ///     The EnsureUserPermissionForContent attribute will deny access to this method if the current user
+    ///     does not have Publish access to this node.
+    /// </remarks>
+    [Authorize(Policy = AuthorizationPolicies.ContentPermissionPublishById)]
+    public IActionResult PostPublishByIdAndCulture(PublishContent model)
+    {
+        IContent? foundContent = GetObjectFromRequest(() => _contentService.GetById(model.Id));
+
+        if (foundContent == null)
+        {
+            return HandleContentNotFound(model.Id);
+        }
+
+        var languageCount = _allLangs.Value.Count();
```

Comment:
I'm wondering if this would not be better on top, for performance reasons: we also call `GetObjectFromRequest` in the `PostPublishById` method, which means that in the case that we end up calling `PostPublishById`, in the current situation we'll have two calls to `GetObjectFromRequest`, meaning two potential trips to the database to load the data.

If we move this part on top, then we'll have only one call in all situations I think.

Reply:
This makes sense indeed. I've moved the check for publishing all cultures to the top of the method.